### PR TITLE
Enforce that Stormpath Java SDK must be built with Maven 3.3.9+

### LINKED ISSUE #1066: Use Maven Enforcer Plugin to require building with Java 7
http://maven.apache.org/enforcer/maven-enforcer-plugin/

Example from JHipster-generated app:

https://github.com/mraible/jhipster-stormpath-example/blob/master/pom.xml#L477

### PR DESCRIPTION
**Pull request by mraible**
Fixes #1066.

This is to prevent anyone from releasing with Maven < 3.3.9.

### CODE REVIEW

File: ci/before_install.sh [hunk 1]
```diff
@@ -13,10 +13,16 @@ export RELEASE_VERSION="$(xmllint --xpath "//*[local-name()='project']/*[local-n
...
+unzip -qq apache-maven-3.3.9-bin.zip
+export M2_HOME=$PWD/apache-maven-3.3.9
+export PATH=$M2_HOME/bin:$PATH
 
 info "Build configuration:"
 echo "Version:             $RELEASE_VERSION"
 echo "Is release:          ${IS_RELEASE:-false}"
 echo "Build documentation: ${BUILD_DOCS:-false}"
 echo "Running IT tests:    ${RUN_ITS:-false}"
+echo "Running IT tests:    ${RUN_ITS:-false}"
```
Comment (dogeared) [inline ci/before_install.sh]: Is this a duplicated line?

File: ci/before_install.sh [hunk 2]
```diff
@@ -13,10 +13,16 @@ export RELEASE_VERSION="$(xmllint --xpath "//*[local-name()='project']/*[local-n
 export IS_RELEASE="$([ ${RELEASE_VERSION/SNAPSHOT} == $RELEASE_VERSION ] && [ $TRAVIS_BRANCH == 'master' ] && echo 'true')"
 export BUILD_DOCS="$([ $TRAVIS_JDK_VERSION == 'oraclejdk8' ] && echo 'true')"
 export RUN_ITS="$([ $TRAVIS_JDK_VERSION == 'openjdk7' ] && echo 'true')"
+#Install Maven 3.3.9 since Travis uses 3.2 by default
+wget https://archive.apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.zip
+unzip -qq apache-maven-3.3.9-bin.zip
+export M2_HOME=$PWD/apache-maven-3.3.9
```
Comment (dogeared) [inline ci/before_install.sh]: I find it weird that `M2_HOME` is used for both maven 2 and 3. This StackO post seems to indicate that all you need is the path: http://stackoverflow.com/questions/26609922/maven-home-mvn-home-or-m2-home

Can we get rid of `M2_HOME`?

**Comment by mraible** [comment]
@dogeared Good catch. I've fixed these issues and pushed. We'll see if Travis likes my changes.

**Comment by mraible** [comment]
@dogeared Doesn't seem to like my changes because enforcer fails in the logs. I'll revert to using `$M2_HOME` and see if that works.

**Comment by mraible** [comment]
@dogeared CI passes - are you OK with me merging this one to `1.1.x`?

**Pull request merged**